src/test/java/org/snomed/snowstorm/core/data/services/ConceptUpdateHelperTest.java

```diff
@@ -0,0 +1,114 @@
+package org.snomed.snowstorm.core.data.services;
+
+import io.kaicode.elasticvc.api.BranchService;
+import org.junit.jupiter.api.Test;
+import org.snomed.snowstorm.AbstractTest;
+import org.snomed.snowstorm.core.data.domain.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ConceptUpdateHelperTest extends AbstractTest {
+	private static final String BRANCH_A = "MAIN/AAA";
+	private static final String SHORTNAME_A = "SNOMEDCT-AAA";
+	private static final String MODULE_A = "12345678910";
+
+	private static final String BRANCH_AB = "MAIN/AAA/BBB";
+	private static final String SHORTNAME_AB = "SNOMEDCT-BBB";
+	private static final String MODULE_AB = "1234567891011";
+
+	private static final String FINDING_SITE = "363698007";
+	private static final String STOMACH = "69695003";
+
+	private static final String CONCEPT_ID = "12345";
+	private static final Concept CONCEPT = new Concept(CONCEPT_ID, MODULE_A)
+			.addFSN("Pizza (food)")
+			.addDescription(new Description("123456", "Cheese."))
+			.addDescription(new Description("1234567", "Tomato."))
+			.addRelationship(new Relationship("1", Concepts.ISA, Concepts.SNOMEDCT_ROOT))
+			.addRelationship(new Relationship("2", FINDING_SITE, STOMACH));
+
+	@Autowired
+	private BranchService branchService;
+
+	@Autowired
+	private ConceptService conceptService;
+
+	@Autowired
+	private CodeSystemService codeSystemService;
+
+	@Test
+	void saveNewOrUpdatedConcepts_ShouldRestoreDetailsFromParentCodeSystem() throws ServiceException {
+		// Extension A creates a Concept et al
+		givenCodeSystemExists(SHORTNAME_A, BRANCH_A);
+		givenBranchExists(BRANCH_A, MODULE_A);
+		givenConceptExists();
+		givenCodeSystemVersionExists();
+		assertEffectiveTimeAndModuleId(getRelationship(getConcept(BRANCH_A)), 20210131, MODULE_A);
+
+		// Extension B modifies Extension A's Concept by de-activating a Relationship
+		givenCodeSystemExists(SHORTNAME_AB, BRANCH_AB);
+		givenBranchExists(BRANCH_AB, MODULE_AB);
+		whenConceptModified(getConcept(BRANCH_A), false);
+		assertEffectiveTimeAndModuleId(getRelationship(getConcept(BRANCH_AB)), null, MODULE_AB);
+
+		// Extension B modifies Extension A's Concept (again) by reverting their change (inactivating Relationship)
+		whenConceptModified(getConcept(BRANCH_AB), true);
+		assertEffectiveTimeAndModuleId(getRelationship(getConcept(BRANCH_AB)), 20210131, MODULE_A); //Effective Time and Module Id restored from parent.
+	}
```

Comment:
This unit test will fail without the appropriate code changes.

Reply:
GREAT unit test.   As we discussed, just to be belt and braces about it, I'd like you to target an existing unit test which will pass through your code and you do NOT want it to take effect.    So you've proven here that it does what you want, I think it would be good to also think about checking that it DOESN'T do what you DON'T want (if you'll excuse the double negative).    Why don't you see what you can find and I'll also check if we have any examples of components in an unexpected module which are then at risk.